hyperspy/io_plugins/hspy.py

```diff
@@ -696,6 +702,16 @@ def write_signal(signal, group, **kwds):
 
     for axis in signal.axes_manager._axes:
         axis_dict = axis.get_axis_dictionary()
+        if 'legacy' in kwds:
```

Comment:
I am not convinced that this is worth supporting saving to an old format?

Reply:
Legacy support is something @francisco-dlp  had brought up.

Reply:
I think the following is clear enough and significantly better than introducing a new saving argument:
https://github.com/hyperspy/hyperspy/blob/60610f4c400ea648f9a95ab5e0dea3a0be670e74/hyperspy/io_plugins/hspy.py#L127-L132

Reply:
I think supporting legacy saving is not necessary because the following is clear enough and significantly better than introducing a new saving argument:
https://github.com/hyperspy/hyperspy/blob/60610f4c400ea648f9a95ab5e0dea3a0be670e74/hyperspy/io_plugins/hspy.py#L127-L132

Reply:
OK, that makes sense; removed the "legacy" option and related tests.